firmware/lm32/heartbeat.h

```diff
@@ -0,0 +1,6 @@
+
```

Comment:
These frame buffers shouldn't be defined in `heartbeat.h`

You should move the definitions out of the hdmi_inX.c files into the hdmi_inX.h file and then include that. (Same with the pattern one.)